packages/@webex/plugin-meetings/src/meeting/index.ts

```diff
@@ -6756,7 +6843,13 @@ export default class Meeting extends StatelessWebexPlugin {
     let floorRequestNeeded = false;
 
     if (tracks.screenShare?.video) {
-      await this.setLocalShareTrack(tracks.screenShare?.video);
+      await this.setLocalShareVideoTrack(tracks.screenShare?.video);
+
+      floorRequestNeeded = true;
```

Comment:
this will result in SDK sending a floor request twice if someone calls publishTracks() first with just a video share track and then again with just audio share track

Reply:
changed this to !this.isJoined
two publishTracks() calls in very quick succession could still result in two calls for the request depending on how long it takes for share to start, but adding more proper state management for floorGrantPending or a new variable is extra work that I haven't scoped yet

Reply:
yes, but I think it's a fairly simple fix we need here - just change the boolean `this.isSharing` into an enum with 3 values (also rename it because current name is not very precise and can be confusing), so if we have: `this.shareFloorState: 'released' | 'pending' | 'granted' then inside `publishTracks()` you could do `floorRequestNeeded = (this.shareFloorState === 'released`)`, right?

Reply:
fair point. added the enum screenShareFloorState